torch/utils/data/_utils/worker.py

```diff
@@ -177,7 +177,7 @@ def _worker_loop(dataset_kind, dataset, index_queue, data_queue, done_event,
                 continue
             if isinstance(r, _ResumeIteration):
                 # Acknowledge the main process
-                data_queue.put(r)
+                data_queue.put((r, None))
```

Comment:
Should `r` be the data?

Reply:
Well, it doesn't really matter. But I prefer `_ResumeIteration` being the index since integer indices are associated with actual data.